experimental/build.nix

```
@@ -18,8 +18,9 @@ _: {
 
           pkgs.protobuf
           pkgs.haskellPackages.haskell-language-server
-          pkgs.swiPrologWithGui
-        ] ++ config.settings.shell.tools;
+        ]
+        ++ (if pkgs.system == "x86_64-linux" then [ pkgs.swiPrologWithGui ] else [ ])
```

You could use `pkgs.lib.optionals` and `pkgs.stdenv.isLinux`
```suggestion
        ++ (pkgs.lib.optionals pkgs.stdenv.isLinux [ pkgs.swiPrologWithGui ])
```